From bf194d776581cca528571f8b3416a545abf7ad30 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 02 Jun 2019 10:25:52 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: restore the div_factor vals

The IF and MESH PLL divider output 1 factors are still using the -1
values for specifying the clock selector value to reference clock
output.
Since commit 6a29399 ([SW-0] pll: extend the hl_pll_cfg structure),
we have the special field for specifying the clock selector values,
so there is no need for "special" value in div_factor field.
Restore the IF and MESH div_factors to their designed values.

Change-Id: If2a6b4318d3953f3cdf02121da7f01c5c37c3ac3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index d060208..d02ddd1 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -28,7 +28,7 @@
 	}, {
 		.name = "IF_W", .base = mmIF_W_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, -1, 2, 7 },
+		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_DIV_PLL,
@@ -37,7 +37,7 @@
 	}, {
 		.name = "IF_E", .base = mmIF_E_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
-		.div_factor = { 0, -1, 2, 7 },
+		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_DIV_PLL,
@@ -55,7 +55,7 @@
 	}, {
 		.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, -1, 2, 7 },
+		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_DIV_PLL,
@@ -73,7 +73,7 @@
 	}, {
 		.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
-		.div_factor = { 0, -1, 2, 7 },
+		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_DIV_PLL,
